### DISCUSSION TITLE
Zigbee 3 gang buttons - Tasmota rule filtering

### DISCUSSION THREAD

**Discussion by olinorwell** [arendst/Tasmota#12535] [Q&A] [answered]
Hello all,

I'm using the "Tuya Smart Life 3 Gang" switches with a Sonoff Zigbee controller (flashed with Tasmota of course).

Basic functionality is working, the devices are paired, I get the Zigbee messages through to the controller etc.

I have a rule that I want to respond to the press of a button on the Tuya remote. The rule sends out some MQTT messages to turn on lights. This works, however it works too well. Every so often it seems like the Zigbee controller is sending out a message to the device to ask it for its state, my rule matches the response to this up as a button press. I want my rule to ignore this message.

(here is the auto message, sent automatically by the Sonoff Zigbee controller box every few hours it seems)

`09:58:39.023 ZIG: Auto-responder: ZbSend {"Device":"0x5471","Cluster":"0x000A","Endpoint":1,"Response":"000A/0007":678535119}`

`09:58:39.032 MQT: TasmotaZBridge/tele/SENSOR = {"ZbReceived":{"0x5471":{"Device":"0x5471","Name":"Lights3Gang","Cluster":10,"Read":[7],"ReadNames":{"LocalTime":true},"Endpoint":1,"LinkQuality":42}}}`

The normal button presses give these messages:

(button 1)
`TasmotaZBridge/tele/SENSOR = {"ZbReceived":{"0x5471":{"Device":"0x5471","Name":"Lights3Gang","0006!FD":"00","Power":253,"Endpoint":1,"LinkQuality":102}}}`

(button 2)
`TasmotaZBridge/tele/SENSOR = {"ZbReceived":{"0x5471":{"Device":"0x5471","Name":"Lights3Gang","0006!FD":"00","Power":253,"Endpoint":2,"LinkQuality":100}}}`

Note - the only difference between the three buttons is the endpoint. So I need to use that to determine what button was pressed.

My rule is this:

`ZBRECEIVED#0X5471#ENDPOINT=1 performs "BACKLOG Publish sonoff_RoomLight1/cmnd/DIMMER 100; Publish sonoff_RoomLight1/cmnd/POWER ON; Publish SonoffLight20X2/cmnd/DIMMER 100; Publish SonoffLight20X2/cmnd/POWER ON; Publish SonoffLight20X3/cmnd/DIMMER 100; Publish SonoffLight20X3/cmnd/POWER ON;"`

Which as you can see is triggered by: `"ZBRECEIVED#0X5471#ENDPOINT=1"
`
This works, and the button on the remote control turns on the lights.

However... the response to the auto responder message also has Endpoint=1 in it. So my lights come on when that automatic message is responded to. This happened at 3am, and 5am this morning (according to my wife... it didn't wake me!).

How can I filter out this automatic message/response when looking for button presses?

I presumed the solution was looking for the lack of a 'Cluster' or 'Response' field in my real button presses. But I can't see how that is possible. I've read that you can have boolean AND/OR in rules but this functionality has to be compiled in - so would I guess mean starting from scratch? I already have 12 devices setup, I don't really want to do that.

Any advice? Currently this solution is unusable as the auto-response message triggers the rule for the first button.

Many thanks in advance.

Oli

**Comment by sfromis** [comment, marked as answer]
Since only the automatic response includes the "Cluster" key, you could prefix your rule with another:
`on ZbReceived#0x5471#Cluster==10 do break` 
This should break off the rest of the rule set when that condition is met.

I'd also recommend you using features like `ZbNameKey` and maybe `ZbNoPrefix` to get much more readable payloads and rules.

**Reply by olinorwell** [reply]
That's a great idea - I hadn't thought of breaking.

Thanks very much, I think that will resolve the problem. Yes I think I'll do the naming as like you say it removes the need to maintain a separate lookup list.
